### PR TITLE
added notes on video

### PR DESCRIPTION
**Pull request by ashishnnnnn**
-> User can add notes on the single video page..for the respective video.
-> Notes will have it's text and player time for the video.
-> Different video will have different notes.

### CODE REVIEW

File: src/Reducer/VideoNotesReducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { v4 as uuid } from "uuid";
+
+export function VideoNotesReducer(state, action) {
+  const action_type = action.type;
+  const { _id, notes } = action.payload;
+  switch (action_type) {
+    case "ADD_NEW_VIDEO_NOTES":
+      return {
+        ...state,
+        [_id]: [{ _id: uuid(), time: notes.time, text: notes.text }],
```
Comment (lannisterrr) [inline src/Reducer/VideoNotesReducer.js:10]: can also use local storage to save notes.

**Review by lannisterrr — COMMENTED** [review]
Structured code , made some suggesstions.

File: src/Reducer/VideoNotesReducer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      };
+    case "ADD_NOTES_TO_PRESENT_VIDEO_NOTES":
+      return {
+        ...state,
+        [_id]: [
+          ...state[_id],
+          { _id: uuid(), time: notes.time, text: notes.text },
+        ],
+      };
```
Comment (KanekarSnehal) [inline src/Reducer/VideoNotesReducer.js:19]: Liked this way of adding notes

**Review by KanekarSnehal — APPROVED** [review]
Code is well structured
Liked the use of ref for calculating the current time
Liked the use of reducer for storing video notes.
Overall great work you can merge it.

**Pull request merged**